File: signalsdk/src/main/kotlin/com/signalsdk/utils/Logger.kt
package com.signalsdk.utils

import android.util.Log

internal object Logger {
    private const val TAG = "[SignalSDK]"
    @Volatile private var enabled = false

    fun enable() { enabled = true }

    fun log(message: String) {
        if (enabled) Log.d(TAG, message)
    }

    fun error(message: String, throwable: Throwable? = null) {
        if (enabled) Log.e(TAG, message, throwable)
    }
}

File: SignalSDK/reactnativesignalsdk/src/utils/logger.ts
const PREFIX = '[WyntaSDK]';

export const logger = {
  log: (message: string, ...args: unknown[]): void => {
    console.log(`${PREFIX} ${message}`, ...args);
  },
  warn: (message: string, ...args: unknown[]): void => {
    console.warn(`${PREFIX} ${message}`, ...args);
  },
  error: (message: string, ...args: unknown[]): void => {
    console.error(`${PREFIX} ${message}`, ...args);
  },
};
